code/common/api.q

```diff
@@ -72,6 +72,21 @@ search:{[s;c]
 // search function
 s:.api.search[;0b]
 
+// input list of namespaces for exportconfig
+KEYS:`$".",'string each key[`] except `$'.Q.an;
+
+// export the current state of config variables, takes in list of symbols of namespaces (e.g. `.usage`.procs)
+exportconfig:{
+  // selects only variables in inpoutted namespaces
+  :?[.api.f`;
+    ((=;`vartype;enlist`variable);(in;`namespace;enlist x));
+    {x!x}enlist`name;
+  // returns name, value and description
+    `val`descrip!(((';.:);`name);`descrip)
```

Comment:
Whats `(;;,:)` supposed to be? A good tip is that you don't want to see q keywords transformed in parse tree you can treat them as symbols instead:
```
q)parse"select `count i from tab"
?
`tab
()
0b
(,`x)!,(,`count;`i)
```